pentaho-js/src/main/javascript/cdf/components/XactionComponent.js

```diff
@@ -49,7 +49,20 @@ define([
                                   " height=\"100%\"" +
                                   " width=\"100%\" />";
           var iframe = $(xactionIFrameHTML);
-          var url = XactionComponentExt.getCdfXaction(this.path, this.action, this.solution) + "&wrapper=false";
+          var actionIncluded = function(path, action) {
+            //check if path ends with action prefixed with '\' or '/'
+            return (typeof path == "string") && (typeof action == "string")
+                && (path.length > action.length)
+                && (path.lastIndexOf(action) == (path.length - action.length))
+                && ("\\/".indexOf(path.substr(-action.length-1, 1))>=0);
+          };
```

Comment:
This is great. Very compact :-) Still, I'd prefer a more readable version of this code, which relied on common operations such as `endsWith`. I know this `String` method cannot yet be safely used, but we can roll our own.

You could place the following two functions at the end of the file, just before the closing `});` of the AMD module:

```js
/**
 * Determines if `path` ends with `action`, 
 * immediately preceded by a `\` or `/` separator character.
 * 
 * Pattern to test: `<path> === <basepath>[\/]<action>`.
 * 
 * This implementation, in particular, strives to not allocate memory.
 * 
 * @param {*} path - The path.
 * @param {*} action - The action.
 * @return {boolean} `true` if yes; `false`, otherwise.
 */
function pathIncludesAction(path, action) {

  if(typeof path !== "string" || typeof action !== "string") {
    return false;
  }

  if(!endsWith(path, action)) {
    return false;
  }

  var separator = path.substr(-(action.length + 1), 1);
  
  return "\\/".indexOf(separator) >= 0;
}

/**
 * Determines if one string ends with another.
 * 
 * @param {string} test - The string being tested to end with `suffix`.
 * @param {string} suffix - The suffix string.
 * @return {boolean} `true` if `test` ends with `suffix`; `false`, otherwise.
 */
function endsWith(test, suffix) {
  return test === suffix || test.slice(-suffix.length) === suffix;
}
```

Documentation is also a good thing to do, even if sometimes it doesn't seem to add much.
It states our intent and helps to validate our assumptions. It allows future folks to know what was on our mind.

Also, note the proposed name, `pathIncludesAction`, instead of `actionIncluded`. If you really prefer something more like you original formulation, then, maybe something like `isActionIncluded` or `doesPathIncludeAction` would still be better.